txcache/trackedBlock.go

```diff
@@ -31,6 +31,41 @@ func newTrackedBlock(
 	}
 }
 
+func (tb *trackedBlock) createOrUpdateVirtualRecords(
+	session SelectionSession,
+	skippedSenders map[string]struct{},
+	sendersInContinuityWithSessionNonce map[string]struct{},
+	accountPreviousBreadcrumb map[string]*accountBreadcrumb,
+	virtualAccountsByAddress map[string]*virtualAccountRecord,
+) error {
+	for address, breadcrumb := range tb.breadcrumbsByAddress {
+		_, ok := skippedSenders[address]
+		if ok {
+			continue
+		}
+
+		accountState, err := session.GetAccountState([]byte(address))
```

Comment:
If this is not the first block where we encounter a breadcrumb for Alice, I think we should not care about what's in the non-virtual session, on her side.

Reply:
I see you've handled that in a different place :+1: